source/Calamari/Kubernetes/ResourceStatus/ResourceStatusCheckTask.cs

```diff
@@ -72,13 +80,28 @@ public async Task<Result> Run(IEnumerable<ResourceIdentifier> resources, Options
                         definedResourceStatuses,
                         resourceStatuses);
 
+                    //if we have been asked to stop, jump out after the last check
+                    if (stopAfterNextStatusCheck)
+                    {
+                        return result;
+                    }
+
                     await timer.WaitForInterval();
+                    
                 } while (!timer.HasCompleted() && result.DeploymentStatus == DeploymentStatus.InProgress);
 
                 return result;
             }, cancellationToken);
         }
 
+        /// <summary>
+        /// Stops the status check loop after the next resource status checks occur.
+        /// </summary> 
+        public void StopAfterNextResourceCheck()
+        {
+            stopAfterNextStatusCheck = true;    
```

Comment:
Is this overcomplicating? Why not just cancel the current status check in progress, then run "one more" before exiting?

Reply:
Because the current status check is a background loop, so I could cancel the current loop, but then re-running it would just trigger the loop again.

I'll take another look at it

Reply:
I looked into this some more and I'm not sure I can easily make a change here without making more wholesale changes to the structure of those classes and the background loop.

Reply:
👍 all good, we will probably review once LOS is available